configure.ac

```diff
@@ -193,9 +193,13 @@ dnl something different but only have that affect the execution of the results
 dnl of the compile, not change the libraries for the compiler itself.
 dnl
 compilersh="run-compiler"
-echo "CC=\"$CC\"" > $compilersh
-echo "LD_LIBRARY_PATH=\"$LD_LIBRARY_PATH\"" >> $compilersh
-echo 'exec $CC $@' >> $compilersh
+export COMPILERSH_CC="$CC"
+export COMPILERSH_LD_LIBRARY_PATH="$LD_LIBRARY_PATH"
+cat <<\EOF > "$compilersh"
+export "CC=$COMPILERSH_CC"
+export "LD_LIBRARY_PATH=$COMPILERSH_LD_LIBRARY_PATH"
+exec $CC "$@"
+EOF
```

Comment:
I prefer the solution that doesn't use public env for this and expand at the time of creation. If we want to stick with this method, prefixing vars with `CURL_` would be a nice touch to avoid collisions.